### PR TITLE
Adding Color support for Philips Hue

### PR DESCRIPTION
**Pull request by cr**
This is adding color support for the Philips Hue adapter. The demo script has been moved to `tools/scripts/huedemo.sh`.

### CODE REVIEW

**Comment by cr** [comment]
@npark-mozilla, @JohanLorenzo, @isabelrios: This will likely not break tests, but it creates new test fodder: new _Color_ setter/getter for _Extended Color lights_, as well as rudimentary support (_OnOff_) for _Dimmable lights_.

File: src/adapters/philips_hue/lights.rs [hunk 1]
```diff
@@ -145,4 +250,41 @@ impl Light {
         self.api.set_light_power(&self.light_id, on);
     }
 
+    #[allow(dead_code)]
+    pub fn get_brightness(&self) -> f64 {
+        let ls = self.api.get_light_status(&self.light_id);
+        ls.state.bri as f64 / 254f64
```
Comment (fabricedesre) [inline src/adapters/philips_hue/lights.rs:257]: why are you dividing by 254 ?

**Comment by fabricedesre** [comment]
@cr, that looks fine overall but can you rebase? I'd like to test locally.

File: src/adapters/philips_hue/lights.rs (hunk 1 above)
Comment (cr) [inline src/adapters/philips_hue/lights.rs:257]: Philips decided to make 254 the maximum value there.

**Comment by cr** [comment]
@fabrice, the rebase is out. Feel free to pull.

File: src/adapters/philips_hue/lights.rs (hunk 1 above)
Comment (fabricedesre) [inline src/adapters/philips_hue/lights.rs:257]: Ok. Can you add a comment to this effect, since it's unusual?

Comment (cr) [inline src/adapters/philips_hue/lights.rs:257]: Done.

**Comment by fabricedesre** [comment]
Works just fine, r=me
thanks!

**Pull request merged**